app/src/main/scala/org/alephium/explorer/service/BlockFlowSyncService.scala

```diff
@@ -137,14 +137,10 @@ case object BlockFlowSyncService extends StrictLogging {
     blockFlowClient: BlockFlowClient,
     cache: BlockCache,
     groupSetting: GroupSetting): Future[Int] = {
-    blockFlowClient.fetchBlocks(from, to, uri).flatMap {
-      case Right(multiChain) =>
-        Future
-          .sequence(multiChain.map(insertBlocks))
-          .map(_.sum)
-      case Left(error) =>
-        logger.error(error)
-        Future.successful(0)
+    blockFlowClient.fetchBlocks(from, to, uri).flatMap { multiChain =>
+      Future
+        .sequence(multiChain.map(insertBlocks))
```

Comment:
Should `insertBlocks` here be transactional? Confirming for #206.

Reply:
yes I think you right, that could be problematic if something fail in the middle of the insert, when we restart we might miss some blocks. 
While syncing from scratch we usually sync around 500 blocks at a time, do you think it could be an issue to use `transitionally`? when it's up-to-dae, it's around 5 blocks at a time so it's fine.

Reply:
thx for the explanation! I think it's fine to use `transactionnally`. the large insert happen only during the `sync-from-scratch` which is usually done "offline".

Reply:
Sounds great 👍🏼

Reply:
I linked the discussion in #206

Reply:
> yes I think you right, that could be problematic if something fail in the middle of the insert, when we restart we might miss some blocks.

Just going based on your comment here. Not having this being a transaction is a bug as it will lead to invalid database state. Since you've marked it as resolved I guess you ok with merging it.

Reply:
Yes the bug was still there before, so we'll tackle it in a new PR.

Reply:
I'll probably work on it today